engines/instance_verification/lib/instance_verification/engine.rb

```diff
@@ -100,6 +100,11 @@ def verify_extension_activation(product)
               'The product is not available for this instance'
             )
           end
+        rescue InstanceVerification::Exception
+          # could not find a subscription for a non-free extension/add-on product
+          # it may require a token in order to be activated via SCC
+          # if no token provided, raise the exception
+          raise unless params.key?(:token) && !params[:token].empty?
```

Comment:
Why does this method getting called at all in the case of a byos/hybrid activation? 
Would it be possible to rename it to `verify_payg_extension_activation!`, and not calling it at all in byos/hybrid case?

Reply:
@digitaltom I am not certain I fully understand the question, but in the hybrid case we do need to check that the underlying base product is a PAYG system.

Reply:
It appears that the logic could be changed here.

In line 94 we raise and that Exception would get caught immediately on line 103 where we then check if there is a token. And we raise potentially on 99 to be caught immediately on line 103.

Appears to me we should check the token condition first, i.e.
```
if ! (params.key?(:token) && !params[:token].empty?)
``` 
and then have the existing code be inside this if condition. Or am I missing something?

Reply:
This method (`verify_extension_activation`), currently only handles payg extension activations, by checking if the extension product is included in the system's PAYG base product subscription. That's why I proposed to rename it to reflect this. 

In the hybrid case, we have a payg system and a requested product activation with provided registration code. I think the presence of the registration code is the trigger there to go into a different code path that forwards the activation request to SCC. And not go into the payg verification, raise and exception, and then hide the exception in the rescue block.